clinch_strike_feature.py

```diff
@@ -0,0 +1,374 @@
+
+    def compute_clinch_strikes_percentages(self, fighter_a_clinch_strikes_hit, fighter_a_clinch_strikes_taken, fighter_b_clinch_strikes_hit, fighter_b_clinch_strikes_taken, differential=False):
+        """
+        Computes the clinch strikes percentages for each fighter in the dataset
+
+        Parameters:
+        - fighter_a_clinch_strikes_hit (float): The number of clinch strikes hit by fighter a
+        - fighter_a_clinch_strikes_taken (float): The number of clinch strikes taken by fighter a
+        - fighter_b_clinch_strikes_hit (float): The number of clinch strikes hit by fighter b
+        - fighter_b_clinch_strikes_taken (float): The number of clinch strikes taken by fighter b
+        - differential (bool): Whether to compute the differential or not
+
+        Returns:
+        - (float, float): The clinch strikes percentages for each fighter as a tuple
+        """
+        fighter_a_clinch_strikes = fighter_a_clinch_strikes_hit / fighter_a_clinch_strikes_taken if fighter_a_clinch_strikes_taken != 0 else 0
+        fighter_b_clinch_strikes = fighter_b_clinch_strikes_hit / fighter_b_clinch_strikes_taken if fighter_b_clinch_strikes_taken != 0 else 0
+
+        if differential:
+            return fighter_a_clinch_strikes - fighter_b_clinch_strikes, fighter_b_clinch_strikes - fighter_a_clinch_strikes
+        else:
+            return fighter_a_clinch_strikes, fighter_b_clinch_strikes
+        
+    def compute_clinch_strikes(self, df, fighter_a_id, fighter_b_id, index, round_number, last_fights, column_name, differential=False, is_defense=False, is_percentage=False):
+        """
+        Computes the clinch strikes for each fighter in the dataset based on column_name
+
+        Parameters:
+        - df (pd.Dataframe): The original dataframe containing all the fights
+        - fighter_a_id (int): The id of the first fighter
+        - fighter_b_id (int): The id of the second fighter
+        - index (int): The index of the current fight
+        - round_number (int): The round number of the current fight
+        - last_fights (int): The number of last fights to consider
+        - column_name (string): The name of the column to compute
+        - differential (bool): Whether to compute the differential or not
+        - is_defense (bool): Whether to compute the defense or not
+        - is_percentage (bool): Whether to compute the percentage or not
+
+        Returns:
+        - (float, float): The clinch strikes for each fighter as a tuple
+        """
+        all_prev_fights = df.loc[:index-1]
+        
+        if not all_prev_fights.empty:
+            fighter_a_id_vals = all_prev_fights.fighter_a_id.values
+            fighter_b_id_vals = all_prev_fights.fighter_b_id.values
+
+            fighter_a_fights = all_prev_fights[(fighter_a_id_vals == fighter_a_id)][-last_fights:]
```

Comment:
Same thing as the other PR - should be this

`fighter_a_fights = all_prev_fights[(fighter_a_id_vals == fighter_a_id) | (fighter_b_id_vals == fighter_a_id)][-last_fights:]`

Reply:
done!